### ISSUE TITLE
[a11y]: Violations found on UI Shell w/ SideNav component in v10 and v11

### ISSUE THREAD

**Issue by LauKwaiFanHK** [carbon-design-system/carbon#11987]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v10, v11

### Automated testing tool and ruleset

IBM Accessibility Checker - Latest Deployment

### Description

I used IBM Accessibility Checker to scan a newly developed page based on React which uses the UI Shell w/ SideNav component. When I scan this component independently (the page on storybook), violations are reported.

### WCAG 2.1 Violation

1.3.1 Info and Relationships, 2.4.1 Bypass Blocks, 4.1.2 Name, Role, Value (v10) and 1.3.1 Info and Relationships, 4.1.2 Name, Role, Value (v11). 

### Reproduction/example

Violations are found on Carbon-React storybook for v10 and v11

### Steps to reproduce

To reproduce violations on v10:
1. Go to https://v10-react.carbondesignsystem.com/iframe.html?id=components-ui-shell--header-base-w-side-nav&viewMode=story&args=
2. Scan this page using IBM Accessibility Checkers
3. 8 violations of 4.1.2 Name, Role, Value will be found:
![Screenshot 2022-08-22 at 14 52 58](https://user-images.githubusercontent.com/81176980/185926786-b433cc4a-f2fc-47ab-8370-3fb18eafe35c.png)

To reproduce violations on v11:
1. Go to https://react.carbondesignsystem.com/iframe.html?viewMode=story&id=components-ui-shell--header-base-w-side-nav&args=
2. Scan this page using IBM Accessibility Checkers
3. 5 violation
![Screenshot 2022-08-22 at 14 53 02](https://user-images.githubusercontent.com/81176980/185926815-bd056c57-0c40-4456-88e7-6bedad02da3e.png)
s of 2.4.1 Bypass Blocks will be found

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by LauKwaiFanHK** [comment]
Besides the violations, particularly high amount of warnings are found for the critieria 2.4.7. Please can you also verify that 2.4.7 is fulfilled? The error states 'Check the keyboard focus indicator is highly visible when using CSS declaration for 'border' or 'outline''.
